openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/DataProductRepository.java

```diff
@@ -91,6 +95,18 @@ public void restorePatchAttributes(DataProduct original, DataProduct updated) {
     updated.withDomain(original.getDomain()); // Domain can't be changed
   }
 
+  @Override
+  protected void postUpdate(DataProduct original, DataProduct updated) {
+    super.postUpdate(original, updated);
+    Map<String, EntityReference> assetsMap = new HashMap<>();
+    listOrEmpty(original.getAssets()).forEach(asset -> assetsMap.put(asset.getId().toString(), asset));
+    listOrEmpty(updated.getAssets()).forEach(asset -> assetsMap.put(asset.getId().toString(), asset));
+    for (EntityReference assetRef : assetsMap.values()) {
+      EntityInterface asset = Entity.getEntity(assetRef, "*", Include.ALL);
```

Comment:
why are we making this call?

Reply:
This is going to penalize the post/put operations on data product as we are making additional calls. Also it looks like this will be triggered on any update to the data product such as description updates , ownership cc @mohityadav766

Reply:
@harshach , beacuse of this i was thinking and added it to the only in entitySpicificUpdates while doing updateAssets to avoid unnecessary calls. in this as we need to update the indexes of assets which are added, removed or update.